pkg/controller/notification/snssubscription/controller.go

```diff
@@ -0,0 +1,243 @@
+	if !reflect.DeepEqual(current, &cr.Spec.ForProvider) {
+		if err := e.kube.Update(ctx, cr); err != nil {
+			return managed.ExternalObservation{},
+				errors.Wrap(err, errKubeSubscriptionUpdateFailed)
+		}
+	}
+
+	// GenerateObservation for SNS Subscription
+	cr.Status.AtProvider = snsclient.GenerateSubscriptionObservation(res.Attributes)
+
+	// Set Status for SNS Subcription
+	switch *cr.Status.AtProvider.Status {
+	case v1alpha1.ConfirmationSuccessful:
+		cr.Status.SetConditions(runtimev1alpha1.Available())
+	default:
+		cr.Status.SetConditions(runtimev1alpha1.Creating())
+	}
+
+	upToDate := snsclient.IsSNSSubscriptionUpToDate(cr.Spec.ForProvider, &sub, res.Attributes)
+	return managed.ExternalObservation{
+		ResourceExists:   true,
+		ResourceUpToDate: upToDate,
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha1.SNSSubscription)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	input := snsclient.GenerateSubscribeInput(&cr.Spec.ForProvider)
+	res, err := e.client.SubscribeRequest(input).Send(ctx)
+
+	if err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, errCreate)
+	}
+
+	meta.SetExternalName(cr, aws.StringValue(res.SubscribeOutput.SubscriptionArn))
+	if err := e.kube.Update(ctx, cr); err != nil {
+		return managed.ExternalCreation{}, errors.Wrap(err, "Failure during kube update")
+	}
+
+	return managed.ExternalCreation{}, nil
+}
+
+func (e *external) Update(ctx context.Context, mgd resource.Managed) (managed.ExternalUpdate, error) {
+	cr, ok := mgd.(*v1alpha1.SNSSubscription)
+	if !ok {
+		return managed.ExternalUpdate{}, errors.New(errUnexpectedObject)
+	}
+
+	// Fetch Subscription Attributes again
+	resp, err := e.client.GetSubscriptionAttributesRequest(&awssns.GetSubscriptionAttributesInput{
+		SubscriptionArn: aws.String(meta.GetExternalName(cr)),
+	}).Send(ctx)
+	if err != nil {
+		return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+	}
+	// Update Subscription
+	attrs := snsclient.GetChangedSubAttributes(cr.Spec.ForProvider, resp.Attributes)
+	for k, v := range attrs {
+		_, err := e.client.SetSubscriptionAttributesRequest(&awssns.SetSubscriptionAttributesInput{
+			AttributeName:   aws.String(k),
+			AttributeValue:  aws.String(v),
+			SubscriptionArn: aws.String(meta.GetExternalName(cr)),
+		}).Send(ctx)
+		if err != nil {
+			return managed.ExternalUpdate{}, errors.Wrap(err, errUpdate)
+		}
+	}
+
+	return managed.ExternalUpdate{}, nil
+}
+
+func (e *external) Delete(ctx context.Context, mgd resource.Managed) error {
+	cr, ok := mgd.(*v1alpha1.SNSSubscription)
+	if !ok {
+		return errors.New(errUnexpectedObject)
+	}
+
+	if !awsarn.IsARN(meta.GetExternalName(cr)) && *cr.Status.AtProvider.Status == v1alpha1.ConfirmationPending {
+		return errors.New("Can't delete a subscription in PendingConfirmation state")
```

Comment:
Should this error string be a constant like the others? Also, I don't think it should start with a capital per https://github.com/golang/go/wiki/CodeReviewComments#error-strings.